### ISSUE TITLE
b-table checkbox

### ISSUE THREAD

**Issue by TanvirAlam** [buefy/buefy#1210]
### Overview of the problem

**Buefy** version: [0.7.1]
**Vuejs** version: [2.5.21]
**OS/Browser**: Chrome

### Description
I have array which is in a table:

```
mainArray = [
 {
   'id' => 23,
   'value' => 34545
 },
 {
   'id' => 24,
   'value' => 745645
 },
 {
   'id' => 25,
   'value' => 456456
 }
]

```

I have checkbox which triggers:

```
<b-table
        :data="Object.values(expenses)"
        :checked-rows.sync="checkedRows"
        :is-row-checkable="(row) => row.status !== 'Paid'"
        checkable
      >
</b-table>
```
I have `watcher`:

```
watch: {
    checkedRows: function(newVal, oldVal) {
        this.updateValue(newVal);
    }
  }
```
with the following method is possible to make the value `0`:

```
updateValue({ getters }, param) {
 state.mainArray.forEach((e1) => {
    param.map(function (selectedValue) {
        if (selectedValue.id == e1.id) {
            Vue.set(selectedValue, "value", 0);
        }
    })
 });
}
```

I do use `_.cloneDeep()` to clone the original array and update the clone one and keep the original intact.

How can do it in one method that once the watcher `checkedRows` is updated, a checkbox is `checked` the value is `0` and `uncheck` return to the previous value?

**Comment by Loruz** [comment]
If I get it right , when you unselect all of your checkboxes, `param` variable becomes empty array, so in `updateValue` there is nothing to iterate on `.map()` and change value back. So you probably need `map` through `mainArray` (if it's a `_.cloneDeep`). Then check if this array contains items from `checkedRows` if so, change that cloned array item value to 0, else -> item value should be equal to original intact arrays item value.
In short, you should have one immutable original array and one copy of that array but with mutable items. It that makes sense

**Comment by TanvirAlam** [comment]
@Loruz 
Is this what you are talking about? 
```
state.mainArray.forEach((e1) => {
    param.map(function (selectedValue) {
        if (selectedValue.id == e1.id) {
            Vue.set(selectedValue, "value", 0);
        }
    })
 });
```

```
state.mainArray.forEach((e1) => {
    param.map(function (selectedValue) {
        if (selectedValue.id == e1.id) {
            Vue.set(selectedValue, "value", 0);
        } else {
                   return back:
        }
    })
 });
```
Its not working: because `:checked-rows.sync="checkedRows"` when `unchecked` it return empty object hence nathing to compared to

**Comment by Loruz** [comment]
```javascript
var mainArray = [
 {
   id : 23,
   value : 34545
 },
 {
   id : 24,
   value : 745645
 },
 {
   id: 25,
   value: 456456
 }
]

var copyOfMain = _.cloneDeep(mainArray)

var param = [{
  id: 23,
  value: 34545
},
 {
   id : 24,
   value : 745645
 }]
 
copyOfMain.map((item, index) => {
  if (param[index] && item.id === param[index].id) {
    return item.value = 0
  } else {
    return item.value = mainArray[index].value
  }
})
```

I guess something like this

**Comment by TanvirAlam** [comment]
@Loruz looks really promising, i will try this tonight. thank you

**Comment by TanvirAlam** [comment]
@Loruz it is not working, the value is not returned back. This is because `param[index]` is empty when `unchecked`, but when `checked` it does turn the value to `0`.

Please advice, maybe I am doing something wrong

**Comment by Loruz** [comment]
Maybe I didn't get what array you want to toggle, or I'm missing some implementation details
Because ` :data="Object.values(expenses)"` i think is not the same as  `copyOfMain`. 
What is `checkedRows` by default ? empty array ? just for reference of checked items? Do you use this `copyOfMain` somewhere in the code after it mutates ? And maybe you nee to try `spread copy` for reactivity. Check here (https://jsbin.com/gudugeq/edit?js,console)

**Comment by TanvirAlam** [comment]
@Loruz `:data="Object.values(expenses)"` basically an array similar to the one

**Comment by TanvirAlam** [comment]
@Loruz Basiclly I am trying to solve

if take look at this: https://codepen.io/pen?&editable=true&editors=101

check change the value of an array and uncheck retrun the value back. take any

**Comment by Loruz** [comment]
@TanvirAlam wrong codepen link. It opens blank empty codepen.

**Comment by TanvirAlam** [comment]
@Loruz please go from here: https://buefy.github.io/documentation/table/

Go to the section: `Checkable`, there you will see CodePen

**Comment by Loruz** [comment]
So you want to change value back and forward only for items that are in checkedRows array ?

**Comment by TanvirAlam** [comment]
@Loruz exactly and also pass event and so all the depended array is also updated

**Comment by Loruz** [comment]
@TanvirAlam  how are you getting `:data="Object.values(expenses)"` and what structure is this `expenses` array

**Comment by TanvirAlam** [comment]
@Loruz 
![peek 2019-02-07 15-14](https://user-images.githubusercontent.com/3250040/52416932-04ef8d00-2aeb-11e9-9cd6-d9af82f266ea.gif)

**Comment by TanvirAlam** [comment]
@Loruz this is what is happening:
![henrik](https://user-images.githubusercontent.com/3250040/52417005-2ea8b400-2aeb-11e9-8e0c-6623573f1508.gif)

**Comment by Loruz** [comment]
@TanvirAlam https://codepen.io/anon/pen/qgpdWj is this what you expect ?

**Comment by TanvirAlam** [comment]
@ if we could send value when `unchecked`, the problem would be solved,

`unchecked`:
```
[__ob__: Observer] 
```

`checked`:
```
(2) [{…}, {…}, __ob__: Observer]
```

**Comment by TanvirAlam** [comment]
@Loruz going back to raw `forloop` I tried that once, never had clue to keep a `flag`. so stupid of me. I am so sorry to waste your time on this. I made it happen with raw manual `checkbox`. 

It works, I would like to buy you a drink or coffee, I have no idea where you at. This solution made my code less number of lines.

Thank you

**Comment by Loruz** [comment]
@TanvirAlam no problem! Have a nice day. It was a pleasure.

**Comment by jtommy** [comment]
@Loruz thanks for the support
